oio/directory/meta2.py

```diff
@@ -0,0 +1,374 @@
+            dst = self._check_dst_service(base, src, dst)
+
+            self.logger.debug(
+                "Moving base (base=%s src=%s dst=%s)", base, src, dst)
+
+            try:
+                """Step 1 of base reassignation algorithm."""
+                peers_to_copy = self._has_base(base, src, dst)
+            except Exception as exc:
+                self.logger.error(
+                    "Failed to check if each peer exists (base=%s src=%s "
+                    "dst=%s): %s", base, src, dst, exc)
+                raise
+            try:
+                """Step 2 of base reassignation algorithm."""
+                self._set_peers(base, src, dst)
+            except Exception as exc:
+                self.logger.error(
+                    "Failed to set new peers (base=%s src=%s dst=%s): %s",
+                    base, src, dst, exc)
+                raise
+            try:
+                """Step 3 of base reassignation algorithm."""
+                self._copy_base(base, src, dst, peers_to_copy)
+            except Exception as exc:
+                self.logger.error(
+                    "Failed to copy base (base=%s src=%s dst=%s): %s",
+                    base, src, dst, exc)
+                raise
+            try:
+                """Step 4 of base reassignation algorithm."""
+                self._link_service(base, src, dst)
+            except Exception as exc:
+                self.logger.error(
+                    "Failed to link service (base=%s src=%s dst=%s): %s",
+                    base, src, dst, exc)
+                raise
+            try:
+                """Step 5 of base reassignation algorithm."""
+                self._reset_election(base, src, dst)
+            except Exception as exc:
+                self.logger.error(
+                    "Failed to reset election (base=%s src=%s dst=%s): %s",
+                    base, src, dst, exc)
+                raise
+        except Exception as exc:
+            self.logger.error(
+                "Failed to move base (base=%s src=%s dst=%s): %s",
+                base, src, dst, exc)
+            err = exc
+        return base, src, dst, err
+
+    def move(self, base, src, dst=None):
+        self.reset_peers()
+
+        try:
+            bases_with_error = self._load_peers(base)
+            if bases_with_error is None:
+                raise ValueError("No peers")
+        except Exception as exc:
+            self.logger.error(
+                "Failed to load peers (base=%s src=%s dst=%s): %s",
+                base, src, dst, exc)
+            yield {'base': base, 'src': src, 'dst': dst, 'err': exc}
+            return
+
+        for base, err in bases_with_error:
+            if err is None:
+                base, src, dst, err = self._safe_move(base, src, dst)
+                yield {'base': base, 'src': src, 'dst': dst, 'err': err}
+            else:
+                yield {'base': base, 'src': src, 'dst': dst, 'err': err}
```

Comment:
`dst` may be still `None`, is it ok ?

Reply:
It's ok.
However, you put your finger on a small duplication of the code.

Reply:
In fact, `dst` should be reset to avoid reusing previous `dst` ?